sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/EntityHelper.java

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.data.tables;
+
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.data.tables.models.TableEntity;
+
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.stream.Collectors;
+
+final class EntityHelper {
+    private static final HashSet<String> TABLE_ENTITY_METHODS = Arrays.stream(TableEntity.class.getMethods())
+        .map(Method::getName).collect(Collectors.toCollection(HashSet::new));
+
+    private EntityHelper() {
+    }
+
+    static void setPropertiesFromGetters(TableEntity entity, ClientLogger logger) {
+        Class<?> myClass = entity.getClass();
+        if (myClass == TableEntity.class) {
+            return;
+        }
+
+        for (Method m : myClass.getMethods()) {
+            // Skip any non-getter methods
+            if (m.getName().length() < 3
+                || TABLE_ENTITY_METHODS.contains(m.getName())
+                || (!m.getName().startsWith("get") && !m.getName().startsWith("is"))
+                || m.getParameterTypes().length != 0
+                || void.class.equals(m.getReturnType())) {
+                continue;
+            }
+
+            int prefixLength = m.getName().startsWith("get") ? 3 : 2;
```

Comment:
Can we could add some documentation around this method. I get 3 is because it's get, but 2 because `is`?

Reply:
I think documentation in general would be nice (even though this is package-private) to facilitate understanding the code base for future devs.

Reply:
Yep, correct. And I fully agree with that, I will add some docs. There is definitely some dark reflection magic here.

Reply:
Resolved